tools/codegen/model.py

```diff
@@ -98,13 +98,15 @@ class NativeFunction:
     # registrations don't participate in codegen-based selective build!
     manual_kernel_registration: bool
 
-    # Distinguish between a missing dispatch dict (historically, this
-    # means to register a catch-all kernel) and a present but empty
-    # dispatch dict (this means register nothing; arguably, this should
-    # subsume manual_kernel_registration).
+    # A mapping of dispatch keys to names of functions implementing
+    # them.  In native_functions.yaml, the dispatch entry is optional; in that
+    # case, that is equivalent to having written:
+    #
+    #   dispatch:
```

Comment:
Is it worth deleting `Math` from all of the native_functions.yaml entries for consistency? Tentatively, I think it making it an implementation detail that Math is the default dispatch entry makes more sense than explicitly offering two options for how people should register ops in native_functions.yaml.

Reply:
I see only one Math entry, in a situation where it is mandatory:

```
- func: native_group_norm(Tensor input, Tensor? weight, Tensor? bias, int N, int C, int HxW, int group, float eps) -> (Tensor, Tensor, Tensor)
  use_c10_dispatcher: hacky_wrapper_for_legacy_signatures
  dispatch:
    CPU, CUDA: native_group_norm
    Math: math_group_norm
```

So we could enforce a canonical form where, if you only have a single Math entry, and its name matches the function schema name, then we can error saying that the Math entry is unnecessary. IDK if it's worth or not.

Reply:
Ah yeah, I didn't notice that some ops require it (since they have other more specific dispatch keys). Probably not worth it then